### PR TITLE
Dtedder/366 android permissions

### LINKED ISSUE #366: Update Android permissions for local files to use Qt 6 Permissions APIs
# Summary
The v.next branch is updated to use Maps SDK 200 and Qt 6. It builds on desktops, but there is still work to do to make Android build. This is due to usage of private permissions APIs that were removed with Qt 6. We should now update Android apps to use data sandboxed to the local app versus using a global `/sdcard` path

# Definition of done
- [x] DSA app makes use of new Permissions APIs
- [x] App builds for Android

### PR DESCRIPTION
**Pull request by dtedder**
Data should be uploaded to the application folder. On Tiramisu, that is:
/data/data/com.esri.arcgisruntime.opensourceapps.DSA_Handheld_Qt/files/ArcGIS/Runtime/Data/DSA

No specific permissions are necessary to access this location.

fixes #366

### CODE REVIEW

File: Shared/messages/MessageFeedsController.h
Comment (JamesMBallard) [inline Shared/messages/MessageFeedsController.h:22]: Ditto try Q_MOC_INCLUDE combined with forward declare.

File: Shared/messages/MessagesOverlay.h
Comment (JamesMBallard) [inline Shared/messages/MessagesOverlay.h:22]: Can you forward declare QString?

File: Handheld/Android/AndroidManifest.xml
Comment (JamesMBallard) [inline Handheld/Android/AndroidManifest.xml:4]: Just a nitpick, but please stick with spaces rather than tabs for consistency.

File: Shared/AddLocalDataController.h
Comment (dtedder) [reply Shared/AddLocalDataController.h:25]: I was actually thrown off by the moc_ include error from the forward declare of QAbstractListModel which like in other cases needed it's own Q_MOC_INCLUDE. My include of DataItemListModel was an incorrect solution to the problem.

File: Shared/RuntimePermissionRequest.cpp
Comment (dtedder) [reply Shared/RuntimePermissionRequest.cpp:21]: Will remove.

File: Shared/alerts/GeoElementAlertTarget.h
Comment (dtedder) [reply Shared/alerts/GeoElementAlertTarget.h:21]: Same. Not sure how this was able to compile.

File: Shared/analysis/AnalysisListController.h
Comment (dtedder) [reply Shared/analysis/AnalysisListController.h:21]: moc includes will be used in the commit to fix.

File: Shared/analysis/LineOfSightController.h
Comment (dtedder) [reply Shared/analysis/LineOfSightController.h:23]: QStringListModel will switch to forward declare. Will moc include the QAbstractItemModel.

File: Shared/analysis/ViewshedController.h
Comment (dtedder) [reply Shared/analysis/ViewshedController.h:21]: same. will moc include

File: Shared/markup/AbstractSketchTool.h
Comment (dtedder) [reply Shared/markup/AbstractSketchTool.h:24]: Again, just thrown off by some other issue building related to the AbstractTool include. This one will switch back to forward declaring the enum class.

File: Shared/messages/MessageFeedsController.h
Comment (dtedder) [reply Shared/messages/MessageFeedsController.h:22]: same as others.

File: Shared/messages/MessagesOverlay.h
Comment (dtedder) [reply Shared/messages/MessagesOverlay.h:22]: QString is not necessary since it's already included by QObject. Removed

File: Handheld/Android/AndroidManifest.xml
Comment (dtedder) [reply Handheld/Android/AndroidManifest.xml:4]: Will do.

**Comment by JamesMBallard** [comment]
The latest changes are looking good @dtedder. We've asked @tanneryould to test drive the branch and double check things are working as expected to get another set of eyes on it. Let's hold off merging until that is complete.

**Comment by JamesMBallard** [comment]
Looks like some minor tweaks are needed to https://github.com/Esri/dynamic-situational-awareness-qt/tree/dtedder/366_android_permissions/docs#app-configuration-settings as well with regards to usage of local data on Android. Please take a look @dtedder.

**Comment by tanneryould** [comment]
I'm able to get the data loaded on the device and have the Android app run on 200.2 with Qt 6.5.2. There's some weirdness with the elevation at further levels of detail -- almost like the slpk has an incorrect vertical offset or something (see [here](https://github.com/Esri/dynamic-situational-awareness-qt/assets/48941951/73818392-8862-46a8-92a3-d68ad42c0933)).

**Comment by dtedder** [comment]
@tanneryould @JamesMBallard 

I pushed an update for the docs/README.md where sdcard was still referenced. Let me know if it's too wordy. But I figured it warranted a bit more explanation now that the location is no longer in shared/legacy storage.

I did a search of the entire repo but couldn't find any more documentation references to /sdcard. The only other pointer I found for this was within the description of the AGO item for the sample data download. Just leaving a reminder here that it should be addressed with the timing of the merge.
https://www.arcgis.com/home/item.html?id=82ce2d85e21c4326bc072d441b636e5e
image

**Comment by JamesMBallard** [comment]
Thanks @dtedder, I logged https://github.com/Esri/dynamic-situational-awareness-qt/issues/370 for that.

**Comment by JamesMBallard** [comment]
Thanks @dtedder, I logged https://github.com/Esri/dynamic-situational-awareness-qt/issues/370 for that.

**Review by JamesMBallard — APPROVED** [review]

**Pull request merged**
